File: app/src/main/java/com/robotnec/budget/app/adapters/transaction/TransactionListItem.java
package com.robotnec.budget.app.adapters.transaction;

import android.support.v7.widget.RecyclerView;

/**
 * @author zak <zak@swingpulse.com>
 */
interface TransactionListItem {
    int getViewType();

    void bindViewHolder(RecyclerView.ViewHolder viewHolder);

    long getId();
}

File: app/src/main/java/com/robotnec/budget/app/adapters/transaction/TransactionListItem.kt
package com.robotnec.budget.app.adapters.transaction

import androidx.recyclerview.widget.RecyclerView

/**
 * @author zak zak@swingpulse.com>
 */
internal interface TransactionListItem {
    val viewType: Int

    val id: Long

    fun bindViewHolder(viewHolder: RecyclerView.ViewHolder)
}
